client/src/test/java/io/pravega/client/stream/impl/ControllerImplTest.java

```diff
@@ -539,17 +545,49 @@ public void deleteScope(ScopeInfo request, StreamObserver<DeleteScopeStatus> res
             }
         };
 
-        fakeServer = InProcessServerBuilder.forName("fakeserver")
-                .addService(fakeServerImpl)
-                .directExecutor()
+        serverPort = TestUtils.getAvailableListenPort();
+        testGRPCServer = NettyServerBuilder.forPort(serverPort)
+                .addService(testServerImpl)
                 .build()
                 .start();
-        controllerClient = new ControllerImpl(InProcessChannelBuilder.forName("fakeserver").directExecutor());
+        controllerClient = new ControllerImpl(URI.create("tcp://localhost:" + serverPort));
     }
 
     @After
     public void tearDown() {
-        fakeServer.shutdown();
+        testGRPCServer.shutdownNow();
+    }
+
+    @Test
+    public void testKeepAlive() throws IOException, ExecutionException, InterruptedException {
+
+        // Verify that keep-alive timeout less than permissible by the server results in a failure.
+        ControllerImpl controller = new ControllerImpl(NettyChannelBuilder.forAddress("localhost", serverPort)
```

Comment:
Here we are setting the client keepalive value to 10s, but the default is 5 minutes, so we expect it to fail, yes?